R/common.R

```diff
@@ -17,16 +17,40 @@
 fromJSON <- function(x) jsonlite::fromJSON(x, TRUE, FALSE, FALSE)
 toJSON   <- function(x) jsonlite::toJSON(x, auto_unbox = TRUE, digits = NA, null="null")
 
-runJaspResults <- function(name, title, dataKey, options, stateKey, functionCall=name)
-{
+loadJaspResults <- function(name) {
+  jaspResultsModule$create_cpp_jaspResults(name, .retrieveState())
+}
+
+finishJaspResults <- function(jaspResultsCPP, returnKeepList = TRUE) {
+
+  jaspResultsCPP$prepareForWriting()
+
+  newState <- list(
+    figures = jaspResultsCPP$getPlotObjectsForState(),
+    other   = jaspResultsCPP$getOtherObjectsForState()
+  )
+
+  jaspResultsCPP$relativePathKeep <- .saveState(newState)$relativePath
+
+  returnThis <- NULL
+  if (returnKeepList)
+    returnThis <- list(keep=jaspResultsCPP$getKeepList()) #To keep the old keep-code functional we return it like this
+
+  jaspResultsCPP$complete() #sends last results to desktop, changes status to complete and saves results to json in tempfiles
```

Comment:
`finishJaspResults` is now being used in `editImage` right?
So you should not only work around the keeplist but you should only call `complete` for an actual run of an analysis. And `jaspResultsCPP$saveResults()` and `$finishWriting()`  instead.

Reply:
I think I did this now, but let me know if this was not what you had in mind!